ext/cache/sqlite.rs

```diff
@@ -153,48 +159,30 @@ impl Cache for SqliteBackedCache {
   ) -> Result<Option<Rc<dyn Resource>>, AnyError> {
     let db = self.connection.clone();
     let cache_storage_dir = self.cache_storage_dir.clone();
-    let maybe_body_path = tokio::task::spawn_blocking(move || {
-      let response_body_key = if request_response.response_has_body {
-        Some(hash(&request_response.request_url))
-      } else {
-        None
-      };
-      let maybe_response_body = {
-        let db = db.lock();
-        db.query_row(
-          "INSERT OR REPLACE INTO request_response_list
-               (cache_id, request_url, request_headers, response_headers, response_body_key, response_status, response_status_text)
-               VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7)
-               RETURNING response_body_key",
-          (
-            request_response.cache_id,
-            &request_response.request_url,
-            serde_json::to_string(&request_response.request_headers)?,
-            serde_json::to_string(&request_response.response_headers)?,
-            response_body_key,
-            &request_response.response_status,
-            &request_response.response_status_text,
-          ),
-          |row| {
-            let response_body_key: Option<String> = row.get(0)?;
-            Ok(response_body_key)
-          },
-        )?
-      };
-      if let Some(body_key) = maybe_response_body {
-        let responses_dir  = cache_storage_dir.join(request_response.cache_id.to_string()).join("responses");
-          std::fs::create_dir_all(&responses_dir)?;
-        let response_path = responses_dir.join(body_key);
-        Ok::<Option<PathBuf>, AnyError>(Some(response_path))
-      } else {
-        Ok::<Option<PathBuf>, AnyError>(None)
-      }
-    }).await??;
-    if let Some(path) = maybe_body_path {
-      // TODO(@satyarohith): can we create the file in the blocking task above?
-      let file = tokio::fs::File::create(path).await?;
-      Ok(Some(Rc::new(CachePutResource::new(file))))
+    let response_body_key = if request_response.response_has_body {
+      Some(hash(&request_response.request_url))
+    } else {
+      None
+    };
```

Comment:
Currently the hash is just based off the request URL. This still results in a race:

1. `cache.put` with req + resp1 starts
2. `cache.put` with resp1 completes
3. `cache.match` with req. This returns resp1.
4. `cache.put` with req + resp2 starts
5. `cache.match` with req. This returns resp1, but with a corrupt body.

To fix this, we need to use a unique string as the key for the storage file for each request. Maybe just a v4 UUID.

Reply:
Oh, I forgot to fix this. I'll use `format!("{request_url_hash}_{time_in_ns}")` as the key.